lib/elixir/src/elixir_bitstring.erl

```diff
@@ -137,16 +143,22 @@ compute_alignment(_, _, _) -> unknown.
 %% Expands the expression of a bitstring, that is, the LHS of :: or
 %% an argument of the bitstring (such as "foo" in "<<foo>>").
 
-expand_expr(_, {{'.', M1, ['Elixir.Kernel', to_string]}, M2, [Arg]}, Fun, E) ->
+expand_expr(_, {{'.', M1, ['Elixir.Kernel', to_string]}, M2, [Arg]}, Fun, _Context, E) ->
   case Fun(Arg, E) of
     {EBin, EE} when is_binary(EBin) -> {EBin, EE};
     {EArg, EE} -> {{{'.', M1, ['Elixir.String.Chars', to_string]}, M2, [EArg]}, EE}
   end;
-expand_expr(Meta, Component, Fun, E) ->
+expand_expr(Meta, Component, Fun, Context, E) ->
   case Fun(Component, E) of
     {EComponent, _} when is_list(EComponent); is_atom(EComponent) ->
       ErrorE = env_for_error(E),
-      form_error(Meta, ?key(ErrorE, file), ?MODULE, {invalid_literal, EComponent});
+      case Context of
+        default ->
+          form_error(Meta, ?key(ErrorE, file), ?MODULE, {invalid_literal, EComponent});
```

Comment:
Let's call it as `form_error(Meta, ?key(ErrorE, file), ?MODULE, {invalid_literal, Context, EComponent})`.

This way we don't need to match here and then match two types (invalid_literal and invalid_literal_in_concat) later on.